persistence.go

```diff
@@ -97,7 +97,7 @@ func (buf *Buffer) load() error {
 			if err != nil {
 				buf.opt.Logger.Error("failed to decompress zero chunk, skipping", zap.String("path", chunkPath.path), zap.Error(err))
 
-				buf.data = buf.data[:cap(buf.data)]
+				buf.data = buf.data[:len(buf.data)]
```

Comment:
This looks strange. It should be a no-op because you are reslicing to the same length.

Reply:
yep, this code was strange, and it's still strange

Reply:
I actually re-did that, the idea was to re-use whatever slice is returned from decompressor, even on failure